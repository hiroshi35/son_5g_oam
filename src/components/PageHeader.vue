<template>
  <div class="row page-title">
    <div class="col-lg-12">
      <h1 class="page-header">
        {{title}}
        <small>{{smallText}}</small>
      </h1>
    </div>
  </div>
</template>

<script>
  export default {
    props: {
      title: {
        type: String,
        required: true
      },
      smallText: {
        type: String
      }
    }
  }
</script>
<style lang="scss">
  $page-background: #e8eef2;
  $blue: #65a4da;
  $text-color: #7d868c;
  .page-title {
    margin-bottom: 10px;
    background-color: lighten($page-background, 1%);
    border-bottom: 1px solid darken($page-background, 10%);
    h1 {
      padding-top: 7px;
      margin-top: 0 !important;
      padding-left: 5px;
      margin-bottom: 0;
      small{
        font-family: Lato;
        font-weight: 300;
      }
    }
    .breadcrumb {
      margin-top: 7px;
      padding-left: 0;
      margin-bottom: 0;
      background-color: transparent;

      .active {
        color: rgba($text-color, 0.9);

      }
      li {

        a {
          color: $blue;

          &:hover, &:focus {
            color: darken($blue, 8%);
            border: none;
            text-decoration: none;
          }

        }
      }

    }
  }
</style>
