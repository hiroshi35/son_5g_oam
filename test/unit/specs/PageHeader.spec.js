import { shallow } from 'vue-test-utils'
import PageHeader from '@/components/PageHeader.vue'

describe('PageHeader.vue', () => {
  let wrapper
  beforeEach(() => {
    wrapper = shallow(PageHeader,
      {
        propsData: {
          title: 'Dashboard',
          smallText: 'Small Text'
        }
      })
  })
  it('should be vue instance', () => {
    expect(wrapper.isVueInstance()).toBeTruthy()
  })
  it('has the expected html structure', () => {
    expect(wrapper.element).toMatchSnapshot()
  })

  it('has correct props', () => {
    expect(wrapper.props().title).toBe('Dashboard')
    expect(wrapper.props().smallText).toBe('Small Text')
  })
})
