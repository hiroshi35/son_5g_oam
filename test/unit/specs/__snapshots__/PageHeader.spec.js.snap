// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageHeader.vue has the expected html structure 1`] = `
<div
  class="row page-title"
>
  <div
    class="col-lg-12"
  >
    <h1
      class="page-header"
    >
      
      Dashboard
      
      <small>
        Small Text
      </small>
    </h1>
  </div>
</div>
`;
